src/hooks/useStorage.ts

```diff
@@ -1,25 +1,35 @@
-/* eslint-disable no-console */
 import { useState } from 'react';
 
-export default (key: string, initialValue: any) => {
-  const [storedValue, setStoredValue] = useState(() => {
+/**
+ * Custom hook for storing and retrieving values in local storage.
+ *
+ * @template T - The type of the value to be stored.
+ * @param {string} key - The key used to identify the stored value.
+ * @param {T} initialValue - The initial value to be stored if no value is found in local storage.
+ * @returns {[T, (value: T) => void]} - An array containing the stored value and a function to update the stored value.
+ * @example const [language, setLanguage] = useStorage<Language>('lang', 'en');
+ */
+function useStorage<T>(key: string, initialValue: T): [T, (value: T) => void] {
+  const [storedValue, setStoredValue] = useState<T>(() => {
     try {
       const item = typeof window !== 'undefined' && window.localStorage.getItem(key);
-      return item || initialValue;
+      return item ? JSON.parse(item) : initialValue;
```

Comment:
why would you use `JSON.parse` here?

Reply:
This is a pretty generic hook, nothing project-specific, this can even go to "common" package or we should provide it in another way. Parsing and stringifying is ensuring that people store serialisable values to the local storage.

https://stackoverflow.com/questions/73341268/storing-an-array-of-dom-elements-in-local-storage-does-not-work-how-to-fix

Reply:
Cool, let's merge this one. 
About adding the hook to a common package, we can probably think about that. `common` probably is not the place since it has some `javascript` specific functions but maybe in the `ui` package or another package with utils content. Anyway, you can file a ticket for this using the parent epic `Search SDK innovations` to start the discussion.